File: 1854-maximum-population-year/1854-maximum-population-year.java
class Solution {
    public int maximumPopulation(int[][] logs) {
        int[] nums = new int[101];
        for(int i=0;i<logs.length;i++){
            for(int j=logs[i][0];j<logs[i][1];j++){
                nums[j-1950]++;
            }
        }
        int maxVal=0;
        int maxYear=1950;
        for(int i=0;i<101;i++){
            if(maxVal<nums[i]){
                maxVal=nums[i];
                maxYear=i+1950;
            }
        }
        return maxYear;
    }
}

File: 1854-maximum-population-year/1854-maximum-population-year.js
/**
 * @param {number[][]} logs
 * @return {number}
 */
var maximumPopulation = function(logs) {
  logs.sort(function(a,b) {
    return a[1]-b[1]
  })
  let map = new Map();
  for(let i=0;i<logs.length;i++){
    for(let j=logs[i][0];j<logs[i][1];j++){
      if(map.has(j)){
        map.set(j,map.get(j)+1);
      }else{
         map.set(j,1)
      }
    }
  }
  let arr=[]
  for(let s of map){
    arr.push(s);
  }
  let maximumPopulation=0;
  let earliestYear=arr[0][0];
  for(let i=0;i<arr.length;i++){
    if(arr[i][1]>maximumPopulation){
      maximumPopulation=arr[i][1];
      earliestYear=arr[i][0]
    }
  }
  return earliestYear;
};